app/Model/Attribute.php

```diff
@@ -1058,15 +1058,19 @@ public function uploadAttachment($fileP, $realFileName, $malware, $eventId = nul
 		if ($malware) {
 			$execRetval = '';
 			$execOutput = array();
-			exec('zip -j -P infected ' . escapeshellarg($zipfile->path) . ' ' . escapeshellarg($fileInZip->path), $execOutput, $execRetval);
+			exec('zip -j -P infected ' . escapeshellarg($zipfile->path) . ' ' . escapeshellarg($tmpfile->path), $execOutput, $execRetval);
 			if ($execRetval != 0) { // not EXIT_SUCCESS
 				throw new Exception('An error has occured while attempting to zip the malware file.');
 			}
 			$fileInZip->delete(); // delete the original non-zipped-file
```

Comment:
if you change $fileInZip to $tmpfile, 
- do it everywhere (also the delete()
- remove it where it's not relevant anymore (double definition)
- a good habit is to also rename a $tmpfile  variable to something that clearly says what's in it, clarifying this bad variable name immediately.